hpo_demo_setup.sh

```diff
@@ -0,0 +1,310 @@
+			check_err "ERROR: HPO failed to start, exiting"
+		else
+			echo
+			echo "Starting hpo with  ./deploy_hpo.sh -c ${CLUSTER_TYPE} -o ${HPO_DOCKER_IMAGE}"
+			echo
+
+			./deploy_hpo.sh -c "${CLUSTER_TYPE}" -o "${HPO_DOCKER_IMAGE}"
+			check_err "ERROR: HPO failed to start, exiting"
+		fi
+	popd >/dev/null
+	echo "#######################################"
+	echo
+}
+
+###########################################
+#   Start HPO Experiments
+###########################################
+## This is function to start experiments with the provided searchspace json.
+## It can be customized to run for any usecase, by providing the searchspace json / 
+## and modifying Step3 to run the benchmark user needs.
+## Currently, it uses TechEmpower benchmark for the demo.
+function hpo_experiments() {
+
+	SEARCHSPACE_JSON="hpo_helpers/search_space.json"
+	URL="http://localhost:8085"
+	exp_json=$(cat ${SEARCHSPACE_JSON})
+	if [[ ${exp_json} == "" ]]; then
+		err_exit "Error: Searchspace is empty"
+        fi
+	## Get experiment_id from searchspace
+	eid=$(${PY_CMD} -c "import hpo_helpers.utils; hpo_helpers.utils.getexperimentid(\"${SEARCHSPACE_JSON}\")")
+	## Get total_trials from searchspace
+	ttrials=$(${PY_CMD} -c "import hpo_helpers.utils; hpo_helpers.utils.gettrials(\"${SEARCHSPACE_JSON}\")")
+	if [[ ${eid} == "" || ${ttrials} == "" ]]; then
+		err_exit "Error: Invalid search space"
+	fi
+
+	echo "#######################################"
+	echo "Start a new experiment with search space json"
+	## Step 1 : Start a new experiment with provided search space.
+	curl -LfSs -H 'Content-Type: application/json' ${URL}/experiment_trials -d '{ "operation": "EXP_TRIAL_GENERATE_NEW",  "search_space": '"${exp_json}"'}'
+	check_err "Error: Creating the new experiment failed."
+
+	## Looping through trials of an experiment
+	for (( i=0 ; i<${ttrials} ; i++ ))
+	do
+		## Step 2: Get the HPO config from HPOaaS
+		echo "#######################################"
+        	echo
+		echo "Generate the config for trial ${i}"
+		echo
+		sleep 10
+		HPO_CONFIG=$(curl -LfSs -H 'Accept: application/json' "${URL}"'/experiment_trials?experiment_id='"${eid}"'&trial_number='"${i}")
+		check_err "Error: Issue generating the configuration from HPO."
+                echo ${HPO_CONFIG}
+		echo "${HPO_CONFIG}" > hpo_config.json
+
+		## Step 3: Run the benchmark with HPO config.
+		## Output of the benchmark should contain objective function result value and status of the benchmark.
+		## Status of the benchmark supported is success and prune
+		## Output format expected for BENCHMARK_OUTPUT is "Objfunc_result=0.007914818407446147 Benchmark_status=success"
+		## Status of benchmark trial is set to prune, if objective function result value is not a number.
+		echo "#######################################"
+		echo
+	        echo "Run the benchmark for trial ${i}"
+		echo
+		BENCHMARK_OUTPUT=$(./hpo_helpers/runbenchmark.sh "hpo_config.json" "${SEARCHSPACE_JSON}" "$i")
```

Comment:
Looks like this benchmark requires openshift or a kubernetes cluster ?
If so do not start the experiments unless we are connected to a cluster

Reply:
The demo benchmark script uses only minikube to run. We are not giving options to user to give any parameters to run the benchmark.
I updated the usage to mention the cluster type in the demo script is only to start HPO service.